src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -413,6 +481,28 @@ public void AttachToDependencyGraph(DependencyAnalysisFramework.DependencyAnalyz
             graph.AddRoot(ThreadStaticsRegion, "ThreadStaticsRegion is always generated");
             graph.AddRoot(StringTable, "StringTable is always generated");
             graph.AddRoot(DispatchMapTable, "DispatchMapTable is always generated");
+            graph.AddRoot(ModuleGlobalData, "ModuleGlobalData is always generated");
+            graph.AddRoot(ModuleHeader, "ModuleHeader is always generated");
+            graph.AddRoot(ModuleIndirectionCell, "ModuleIndirectionCell is always generated");
+
+            var stringType = _context.GetWellKnownType(WellKnownType.String);
+            if (!_context.IsTypeDefinedExternally(stringType))
```

Comment:
You can get rid of this awkward code once #853 is merged. After you have it, just use `EETypePtr.EETypePtrOf<String>()` in place of `GetStringEEType()`.